### PR TITLE
Remove compatible providers validation for AppCatalogEntry CRD

### PR DESCRIPTION
**Pull request by rossf7**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: config/crd/application.giantswarm.io_appcatalogentries.yaml [hunk 1]
```diff
@@ -143,14 +143,8 @@ spec:
...
                       app is compatible with all providers.
-                    items:
-                      enum:
-                      - aws
-                      - azure
-                      - kvm
-                      type: string
                     nullable: true
-                    type: array
+                    type: string
```
Comment (tfussell) [inline config/crd/application.giantswarm.io_appcatalogentries.yaml:147]: We already have this as a list in `Chart.yaml` (see https://github.com/giantswarm/cluster-openstack/blob/main/helm/cluster-openstack/Chart.yaml#L9-L11). Will this cause problems if we use a string type here?

File: api/v1alpha1/app_catalog_entry_types.go [hunk 2]
```diff
@@ -125,14 +125,11 @@ type AppCatalogEntrySpecRestrictions struct {
...
 	// CompatibleProviders is a list of provider names which this app is compatible with. Default is empty.
 	// Empty list means app is compatible with all providers.
-	CompatibleProviders []Provider `json:"compatibleProviders,omitempty"`
+	CompatibleProviders string `json:"compatibleProviders,omitempty"`
 	// NamespaceSingleton is a flag for whether this app can be installed at most once per namespace. Default is false.
 	NamespaceSingleton bool `json:"namespaceSingleton,omitempty"`
 }
 
-// +kubebuilder:validation:Enum=aws;azure;kvm
-type Provider string
```
Comment (tfussell) [inline api/v1alpha1/app_catalog_entry_types.go:134]: This is still used in the `_test.go` file.

File: config/crd/application.giantswarm.io_appcatalogentries.yaml (hunk 1 above)
Comment (rossf7) [reply config/crd/application.giantswarm.io_appcatalogentries.yaml:147]: Yes it needs to be a list.

**Review by tfussell — APPROVED** [review]

**Comment by rossf7** [comment]
Tested on gamma

**Pull request merged**